biorbd_optim/interfaces/acados_interface.py

```diff
@@ -63,58 +71,74 @@ def __prepare_acados(self, ocp):
-                    raise RuntimeError(
-                        "Initial constraint on state must be hard. Hint: you can pass it as an objective"
-                    )
-                else:
-                    self.acados_ocp.constraints.x0 = np.array(ocp.nlp[i].X_bounds.min[:, 0])
-                    self.acados_ocp.dims.nbx_0 = self.acados_ocp.dims.nx
-
-            # control constraints
-            self.acados_ocp.constraints.constr_type = "BGH"
-            self.acados_ocp.constraints.lbu = np.array(ocp.nlp[i].U_bounds.min[:, 0])
-            self.acados_ocp.constraints.ubu = np.array(ocp.nlp[i].U_bounds.max[:, 0])
-            self.acados_ocp.constraints.idxbu = np.array(range(self.acados_ocp.dims.nu))
-            self.acados_ocp.dims.nbu = self.acados_ocp.dims.nu
-
-            # path constraints
-            self.acados_ocp.constraints.Jbx = np.eye(self.acados_ocp.dims.nx)
-            self.acados_ocp.constraints.ubx = np.array(ocp.nlp[i].X_bounds.max[:, 1])
-            self.acados_ocp.constraints.lbx = np.array(ocp.nlp[i].X_bounds.min[:, 1])
-            self.acados_ocp.constraints.idxbx = np.array(range(self.acados_ocp.dims.nx))
-            self.acados_ocp.dims.nbx = self.acados_ocp.dims.nx
-
-            # terminal constraints
-            self.acados_ocp.constraints.Jbx_e = np.eye(self.acados_ocp.dims.nx)
-            self.acados_ocp.constraints.ubx_e = np.array(ocp.nlp[i].X_bounds.max[:, -1])
-            self.acados_ocp.constraints.lbx_e = np.array(ocp.nlp[i].X_bounds.min[:, -1])
-            self.acados_ocp.constraints.idxbx_e = np.array(range(self.acados_ocp.dims.nx))
-            self.acados_ocp.dims.nbx_e = self.acados_ocp.dims.nx
-
-        return self.acados_ocp
+        self.acados_ocp.dims.nx = ocp.nlp[0].nx
+        self.acados_ocp.dims.nu = ocp.nlp[0].nu
+        self.acados_ocp.dims.ny = self.acados_ocp.dims.nx + self.acados_ocp.dims.nu
+        self.acados_ocp.dims.ny_e = ocp.nlp[0].nx
+        self.acados_ocp.dims.N = ocp.nlp[0].ns
+
+    def __set_constr_type(self, constr_type="BGH"):
+        self.acados_ocp.constraints.constr_type = constr_type
+        self.acados_ocp.constraints.constr_type_e = constr_type
+
+    def __set_constrs(self, ocp):
+        # constraints handling in self.acados_ocp
+        u_min = np.array(ocp.nlp[0].U_bounds.min)
+        u_max = np.array(ocp.nlp[0].U_bounds.max)
+        x_min = np.array(ocp.nlp[0].X_bounds.min)
+        x_max = np.array(ocp.nlp[0].X_bounds.max)
+
+        if not np.all(np.all(u_min.T == u_min.T[0, :], axis=0)):
+            raise NotImplementedError("U_bounds min must be the same at each shooting point with ACADOS")
+        if not np.all(np.all(u_max.T == u_max.T[0, :], axis=0)):
+            raise NotImplementedError("U_bounds max must be the same at each shooting point with ACADOS")
+
+        if np.min(u_min) == -np.inf or np.min(x_min) == -np.inf or np.max(u_max) == np.inf or np.max(x_max) == np.inf:
```

Comment:
I think `not np.isfinite()` is more secured than `== np.inf`
https://numpy.org/doc/stable/reference/generated/numpy.isfinite.html